### PR TITLE
Feature/user authorization

### PR DESCRIPTION
**Pull request by mlshvch**
# Description

Add user authorization. Roles can be assigned to authenticated users.

- Reader: can read news
- Correspondent: can read and create news, update and delete only the news he has created
- Editor: can approve, create, edit and delete all the news 
- Admin: resource administrator

## Trello card link

https://trello.com/c/S4XdP7BE/11-authorization-roles

# Checklist:

- [x] My code fixes the problem or adds required functionality
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove that my feature works
- [x] New and existing unit tests pass locally with my changes

# Additional requirements:

- [x] Does your code require any migrations to be ran?
- [x] Does your code require installing any new gems or system dependencies?
- [ ] Does your code require any changes to current deployment process?

# Additional Information

### CODE REVIEW

**Comment by mlshvch** [comment]
Test are not ready yet.

File: app/controllers/news_articles_controller.rb [hunk 1]
```diff
@@ -22,7 +22,9 @@ def edit; end
 
   # POST /news_articles
   def create
-    @news_article = NewsArticle.new(news_article_params)
+    authorize NewsArticle.new
+    @news_article = policy_scope(NewsArticle).new(news_article_params)
```
Comment (blackp0is0n) [inline app/controllers/news_articles_controller.rb:26]: @news_article = NewsArticle.new(news_article_params)
authorize @news_article

File: app/controllers/news_articles_controller.rb [hunk 2]
```diff
@@ -37,6 +39,8 @@ def create
   def update
     respond_to do |format|
       if @news_article.update(news_article_params)
+        authorize @news_article
```
Comment (blackp0is0n) [inline app/controllers/news_articles_controller.rb:42]: authorise should be before update

File: app/controllers/news_articles_controller.rb [hunk 3]
```diff
@@ -46,7 +50,7 @@ def update
 
   # DELETE /news_articles/1
   def destroy
-    @news_article.destroy
+    policy_scope(NewsArticle).find(params[:id]).destroy
```
Comment (blackp0is0n) [inline app/controllers/news_articles_controller.rb:53]: `authorize @news_article`
`@news_article.destroy`

File: app/controllers/news_articles_controller.rb [hunk 4]
```diff
@@ -2,11 +2,11 @@
 
 class NewsArticlesController < ApplicationController
   before_action :set_news_article, only: %i[show edit update destroy]
-  before_action :authenticate_user!
+  before_action :authenticate_user!, except: :index
```
Comment (blackp0is0n) [inline app/controllers/news_articles_controller.rb:5]: Pls remove this Line and move following:
  ```
  after_action :verify_authorized, except: :index, unless: :devise_controller?
  after_action :verify_policy_scoped, only: :index
  ```
  to application controller

**Review by blackp0is0n — CHANGES_REQUESTED** [review]

**Review by blackp0is0n — APPROVED** [review]

**Pull request merged**